tests/unit/sidecars/live_data_watcher_loader_tests.py

```diff
@@ -77,10 +77,16 @@ def test_load_from_s3_missing_config(start_mock_s3, missing_config):
         assert _load_from_s3(json.dumps(data).encode("utf-8"))
 
 
-def test_successful_load_from_s3(s3_stub):
+@pytest.mark.parametrize(
+    "anon",
+    (
+        True,
+        False,
```

Comment:
I guess because this is a unit test, this really changes how it communicates with s3 and therefore the functionality is not any more testable beyond ensuring that you go down that code path.

🔕 would like to see some other mocked s3 responses or checking that swapping this flag changes the config of how the request is made

Reply:
Yeah I was thinking about how to go about this, but best I could do is just to make test that both anon=True and anon=False get us down the codepath and return content. 

Testing the actual functionality of an s3 API seems beyond unit testing scope here...overall i agree with your sentiment though